### PR TITLE
Fix for HA 2023.4

### LINKED ISSUE #527: Apexcharts breaks on 2023.4.0b0
**Checklist**
- [X] I updated the card to the latest version available
- [X] I cleared the cache of my browser
- [X] I verified that I'm really running the lastest version in my browser console
- [X] I checked if there is another issue opened with the same problem

**Describe the bug**
All my apex charts error out on the beta for 2023.4

image

**Version of the card**
Version: 2.0.2

**To Reproduce**
This is the configuration I used:
https://github.com/hmmbob/HomeAssistantConfig/blob/fda9dd15951d69656aee717ef934901978f92e6e/dashboards/10-energy.yaml#L104-L123

**Screenshots**
See above

**Desktop (please complete the following information):**
Chrome v111

**Smartphone (please complete the following information):**
HA Android Companion app same result

### PR DESCRIPTION
**Pull request by hmmbob**
SSIA.

Fixes #527

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -142,28 +185,37 @@ export function offsetData(data: EntityCachePoints, offset: number | undefined):
...
 export function getLovelace(): LovelaceConfig | null {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let root: any = document.querySelector('home-assistant');
+  let root: any = document.querySelector("home-assistant");
   root = root && root.shadowRoot;
-  root = root && root.querySelector('home-assistant-main');
+  root = root && root.querySelector("home-assistant-main");
   root = root && root.shadowRoot;
-  root = root && root.querySelector('app-drawer-layout partial-panel-resolver');
+  root = root && root.querySelector("mwc-drawer partial-panel-resolver");
```
Comment (hmmbob) [inline src/utils.ts:214]: This is the actual fix

File: hacs.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
-    "name": "apexcharts-card",
-    "render_readme": true
+  "homeassistant": "2023.4.0b0",
```
Comment (hmmbob) [inline hacs.json:2]: Ensure backward compatibility for older versions (or: only show this version of Apexcharts to users with HA2023.4.0b0 or higher)

**Comment by RomRider** [comment]
Thanks, I can't currently merge that, but I'll review it early next week.

**Comment by hmmbob** [comment]
I've changed it to `ha-drawer` directly, given the comment of Bram

File: src/utils.ts [hunk 3]
```diff
@@ -142,28 +185,37 @@ export function offsetData(data: EntityCachePoints, offset: number | undefined):
...
 export function getLovelace(): LovelaceConfig | null {
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  let root: any = document.querySelector('home-assistant');
+  let root: any = document.querySelector("home-assistant");
   root = root && root.shadowRoot;
-  root = root && root.querySelector('home-assistant-main');
+  root = root && root.querySelector("home-assistant-main");
   root = root && root.shadowRoot;
-  root = root && root.querySelector('app-drawer-layout partial-panel-resolver');
+  root = root && root.querySelector("ha-drawer partial-panel-resolver");
```
Comment (hmmbob) [inline src/utils.ts:214]: Actual change is here

**Comment by RomRider** [comment]
I'm closing this in favor of e1aaf691aee8bb4e8c8b75eb3fb301d1c31d20b6

**Pull request closed without merging**